enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,273 @@
+of the webhook is accurate; the issue lies with this required field in the `ExternalDNS` CRD.
+
+However, there exists a simple workaround: OpenShift users can specify an empty `""` `Credentials` field as follows:
+```yaml
+apiVersion: externaldns.olm.openshift.io/v1beta1
+kind: ExternalDNS
+metadata:
+  name: sample-aws
+spec:
+  provider:
+    type: AWS
+    aws:
+      assumeRole:
+        arn: "arn:aws:xxx"
+      credentials:
+        name: ""
+```
+This satisfies the `ExternalDNS` CRD `+required` field while the ExternalDNS Operator logic [ignores](https://github.com/openshift/external-dns-operator/blob/bac092e98fe5f9065b75bd0fa21d5aeff9d00853/pkg/operator/controller/credentials-secret/controller.go#L259-L267)
+the empty string and uses the `CredentialsRequest`. As a result, we plan to introduce this `AssumeRole` field in the
+v1beta1 release accompanied by documentation that specifies this requirement to specify an empty `Credentials` field in
+OpenShift.
+
+In the next version of ExternalDNS Operator's API, we will address the requirement for the `Credentials` field as
+outlined in the alternative [Bumping the API](#bumping-the-api), thus resolving this particular issue.
+
+### Risks and Mitigations
+
+* Introducing this API change in v1beta1 exposes the issue outlined in [The Required Credentials Issue](#the-required-credentials-issue)
+  and has the potential to cause confusion. We plan to mitigate this by documenting the limitation. 
+
+### Drawbacks
+
+No obvious drawbacks as this enhancement is merely exposing an existing ExternalDNS functionality via an API.
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### Test Plan
+
+We will add E2E tests for the ExternalDNS Operator utilizing two AWS CI accounts to create DNS records in one (Account
+A) and install the cluster in the other (Account B). This E2E test will leverage existing steps in the step registry
+as created by [AWS Shared VPC with Cross-account DNS Zones](/enhancements/installer/aws-cross-account-dns-zone.md) such
+as [ipi-aws-pre-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-pre-shared-vpc-phz) and [ipi-aws-post-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-post-shared-vpc-phz).
+These steps set up a shared VPC cluster with access to a shared DNS private hosted zone. However, we may choose to use
+a public hosted zone instead, as there is no functional distinction between the two, and it simplifies the E2E testing
+process.
+
+### Graduation Criteria
+
+This update is an extension of the existing v1beta1 ExternalDNS Operator API and does not impact the graduation
+criteria for the v1beta1 API.
+
+The ExternalDNS Operator updates to support Shared VPC are out of payload and are therefore not aligned with
+OpenShift releases. However, we are targeting to release this enhancement of ExternalDNS as alongside of OCP 4.14.
```

Comment:
This has changed, right?  It would have to be released in 4.15 and backported to 4.14.z.